### PR TITLE
Added: Multiple sources for video/audio/images

### PR DESCRIPTION
**Pull request by KonWol**
Allows specifying multiple sources for video and audio. The browser then chooses the format it supports. For images, only a single resource is supported. A warning is displayed, if not exactly one image resource is provided.

### CODE REVIEW

File: src/questionnaireitem_media_video.js [hunk 1]
```diff
@@ -20,6 +20,8 @@ function QuestionnaireItemMediaVideo(className, question, required, url, readyOn
 
     TheFragebogen.logger.debug(this.constructor.name + "()", "Set: className as " + this.className + ", urls as " + this.height + ", width as " + this.width);
 
+    this.videoSource = null;
+    this.pTag = null;
```
Comment (dennisguse) [inline src/questionnaireitem_media_video.js]: Not requiried as class attribute.
It is only used in createUI().

File: src/questionnaireitem_media_video.js [hunk 2]
```diff
@@ -40,11 +42,28 @@ QuestionnaireItemMediaVideo.prototype._createAnswerNode = function() {
 };
 
 QuestionnaireItemMediaVideo.prototype.releaseUI = function() {
-    this.node = null;
```
Comment (dennisguse) [inline src/questionnaireitem_media_video.js]: Please leave it on top. It will be replaced by a super()-call when transformation to ES6 class structure is done.

File: src/questionnaireitem_media_video.js [hunk 3]
```diff
@@ -40,11 +42,28 @@ QuestionnaireItemMediaVideo.prototype._createAnswerNode = function() {
...
     this.uiCreated = false;
     this.enabled = false;
 
+    this.videoNode.pause();
+    if (Array.isArray(this.videoSource)) {
```
Comment (dennisguse) [inline src/questionnaireitem_media_video.js]: I would recommend making this.videoSource an array by default in the constructor.
It's easier to read.

File: src/questionnaireitem_media_video.js [hunk 4]
```diff
@@ -58,8 +77,25 @@ QuestionnaireItemMediaVideo.prototype._createMediaNode = function() {
     }
 
     this.videoNode = document.createElement('video');
+    this.videoNode.setAttribute("playsinline", "");
```
Comment (dennisguse) [inline src/questionnaireitem_media_video.js]: Please add a comment above why this is done.
And please make a separate commit.

File: src/questionnaireitem_media_video.js [hunk 5]
```diff
@@ -20,6 +20,8 @@ function QuestionnaireItemMediaVideo(className, question, required, url, readyOn
 
     TheFragebogen.logger.debug(this.constructor.name + "()", "Set: className as " + this.className + ", urls as " + this.height + ", width as " + this.width);
 
+    this.videoSource = null;
```
Comment (dennisguse) [inline src/questionnaireitem_media_video.js]: What is the advantage of keeping track of the video sources?
I think removing the node from DOM should remove the sources and clear their memory, or?
If not so, please provide as a separate PR.

**Review by dennisguse — CHANGES_REQUESTED** [review]

File: src/questionnaireitem_media_video.js (hunk 1 above)
Comment (KonWol) [reply src/questionnaireitem_media_video.js]: Removed it.

File: src/questionnaireitem_media_video.js (hunk 2 above)
Comment (KonWol) [reply src/questionnaireitem_media_video.js]: Restored original implementation.

File: src/questionnaireitem_media_video.js (hunk 4 above)
Comment (KonWol) [reply src/questionnaireitem_media_video.js]: Moved to new PR #55

File: src/questionnaireitem_media_video.js [hunk 6]
```diff
@@ -59,7 +59,23 @@ QuestionnaireItemMediaVideo.prototype._createMediaNode = function() {
...
+            videoSource.src = this.url[i];
+            this.videoNode.appendChild(videoSource);
+        }
+    } else {
+        videoSource = document.createElement("source");
+        videoSource.src = this.url;
+        this.videoNode.appendChild(videoSource);
+    }
+
+    pTag = document.createElement("p");
```
Comment (dennisguse) [inline src/questionnaireitem_media_video.js:71]: Please move adding pTag to a separate commit and rename it to videoNotSupportMessage.
Then it's meaning is easier to understand.

File: src/questionnaireitem_media_video.js [hunk 7]
```diff
@@ -59,7 +59,23 @@ QuestionnaireItemMediaVideo.prototype._createMediaNode = function() {
 
     this.videoNode = document.createElement('video');
     this.videoNode.oncanplaythrough = this._onloaded.bind(this);
-    this.videoNode.src = this.url;
+
+    if (Array.isArray(this.url)) {
```
Comment (dennisguse) [inline src/questionnaireitem_media_video.js]: This changes the implementation approach as `this.url` is inherited from QuestionnaireItemMedia and it defined as one url only.
Thus QuestionnaireItemMediaVideo would use this in a different, manner which might be confusing.

I would recommend:
* changing the QuestionnaireItemMedia (and all it's childs), so that multiple URLs are supported (ie. make this.url an Array of string).
* for *Image only allow one URL by providing a warning if more than one are provided and only take the first one.
* In the constructor of QuestionnaireItemMedia: allow url to be string as well as Array of string (update documentation and use `this.url = Array.isArray(url) ? url : [url]`.
In this way it must not be distinguished later.
* update createUI() code for all childs of QuestionnaireItemMedia

**Review by dennisguse — CHANGES_REQUESTED** [review]
See comments.

File: src/questionnaireitem_media_video.js (hunk 6 above)
Comment (KonWol) [reply src/questionnaireitem_media_video.js:71]: okay done.

File: src/questionnaireitem_media_image.js [hunk 8]
```diff
@@ -20,6 +20,10 @@ function QuestionnaireItemMediaImage(className, question, required, url, readyOn
 
     TheFragebogen.logger.debug("QuestionnaireItemMediaImage()", "Set: className as " + this.className + ", height as " + this.height + ", width as " + this.width);
 
+    if (this.url.length != 1) {
+        TheFragebogen.logger.warn("QuestionnaireItemMediaImage()", "called with multiple resources as url. Falling back to the first element in the array.");
```
Comment (dennisguse) [inline src/questionnaireitem_media_image.js:24]: `this.constructor.name` instead of hardcoding the classname.

**Review by dennisguse — CHANGES_REQUESTED** [review]
Again only small nitpicking...
Otherwise it looks nice!

**Review by dennisguse — DISMISSED** [review]
